File: go-part/controller/weather_controller.go
package controller

import (
	"github.com/gin-gonic/gin"
	"go-weather-api/service"
	"net/http"
)

func GetWeather(c *gin.Context) {
	city := c.Query("city")
	if city == "" {
		c.JSON(http.StatusBadRequest, gin.H{"error": "city is required"})
		return
	}

	data, err := service.GetWeatherWithCache(city)
	if err != nil {
		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
		return
	}

	c.JSON(http.StatusOK, data)
}

File: ts-part/src/controllers/weather.controller.ts
import { Request, Response } from 'express';
import { getWeatherWithCache } from '../services/weather.service';

export const getWeather = async (req: Request, res: Response): Promise<any> => {
  try {
    const city = req.query.city as string;
    if (!city) {
      return res.status(400).json({ message: 'City is required' });
    }

    // Call service here
    const data = await getWeatherWithCache(city);
    return res.json(data);
  } catch (err) {
    return res.status(500).json({ message: 'Something went wrong', error: err });
  }
};
